File: web/src/api/admin/vuln.ts
import request from "@/utils/requests"

//列出漏洞类型
export function listVulnType(data:any){
    return request({
        url: '/vuln/listVulnType',
        headers: {
          isToken: true,
          'Content-Type': 'application/json'
        },
        method: 'post',
        data:data
      })
}

//添加漏洞类型数据
export function createVulnType(data:any){
    return request({
        url: '/vuln/createVulnType',
        headers: {
          isToken: true,
          'Content-Type': 'application/json'
        },
        method: 'post',
        data:data
      })
}

//删除漏洞类型数据
export function deleteVulnType(data:any){
    return request({
        url: '/vuln/deleteVulnType',
        headers: {
          isToken: true,
          'Content-Type': 'application/json'
        },
        method: 'post',
        data:data
      })
}

//更新漏洞类型数据
export function updateVulnType(data:any){
    return request({
        url: '/vuln/updateVulnType',
        headers: {
          isToken: true,
          'Content-Type': 'application/json'
        },
        method: 'post',
        data:data
      })
}

//获取树形数据
export function buildVulnTypeTree(data:any){
  return request({
      url: '/vuln/buildVulnTypeTree',
      headers: {
        isToken: true,
        'Content-Type': 'application/json'
      },
      method: 'get',
      data:data
    })
}


//提交漏洞数据
export function submitVuln(data:any){
  return request({
      url: '/vuln/submitVuln',
      headers: {
        isToken: true,
        'Content-Type': 'application/json'
      },
      method: 'post',
      data:data
    })
}
//查看漏洞数据
export function listVuln(data:any){
  return request({
      url: '/vuln/listVuln',
      headers: {
        isToken: true,
        'Content-Type': 'application/json'
      },
      method: 'post',
      data:data
    })
}

//获取漏洞数据
export function getVuln(data:any){
  return request({
      url: '/vuln/getVuln',
      headers: {
        isToken: true,
        'Content-Type': 'application/json'
      },
      method: 'get',
      data:data
    })
}
//更新漏洞数据
export function updateVuln(data:any){
  return request({
      url: '/vuln/updateVuln',
      headers: {
        isToken: true,
        'Content-Type': 'application/json'
      },
      method: 'post',
      data:data
    })
}
//删除漏洞数据
export function deleteVuln(data:any){
  return request({
      url: '/vuln/deleteVuln',
      headers: {
        isToken: true,
        'Content-Type': 'application/json'
      },
      method: 'post',
      data:data
    })
}

//删除漏洞数据
export function auditVuln(data:any){
  return request({
      url: '/vuln/auditVuln',
      headers: {
        isToken: true,
        'Content-Type': 'application/json'
      },
      method: 'post',
      data:data
    })
}

File: backend/routes/vuln.go
package routes

import (
	"github.com/gin-gonic/gin"
	"xiaoyun/backend/controllers/vuln_controller"
	"xiaoyun/backend/middleware"
)

func SetupVulnRoutes(route *gin.Engine) {
	////不需要登录
	//{
	//	r := route.Group("/")
	//	r.GET(apiuri+"/getvulnabs", controllers.GetVulnAbstract)
	//	r.GET(apiuri+"/getvulntypes", controllers.GetVulnTypeList)
	//	r.GET(apiuri+"/getvulntype", controllers.GetVulnType)
	//	r.GET(apiuri+"/getvulnlist", controllers.GetVulnList)
	//	r.GET(apiuri+"/getvulndtl", controllers.GetVulnDetail)
	//}
	////需要登录
	//{
	//	r := route.Group("/")
	//	r.POST(apiuri+"/addvuln", controllers.AddVuln)
	//	r.POST(apiuri+"/editvuln", controllers.EditVuln)
	//	r.GET(apiuri+"/uservulnlist", controllers.GetUserVulnList)
	//}
	{
		r := route.Group(apiuri + "/vuln")
		r.GET("/buildVulnTypeTree", vuln_controller.BuildVulnTypeTree)

	}
	{
		r := route.Group(apiuri + "/vuln")
		r.Use(middleware.AuthMiddleware())
		r.POST("/submitVuln", vuln_controller.SubmitVuln)
	}
	////需要管理员
	{
		r := route.Group(apiuri + "/vuln")
		r.Use(middleware.AuthMiddleware(), middleware.AdminMiddleware())
		r.POST("/listVulnType", vuln_controller.ListVulnType)
		r.POST("/createVulnType", vuln_controller.CreateVulnType)
		r.POST("/updateVulnType", vuln_controller.UpdateVulnType)
		r.POST("/deleteVulnType", vuln_controller.DeleteVulnType)

		//漏洞处理相关
		r.POST("/listVuln", vuln_controller.ListVuln)
		r.POST("/createVuln", vuln_controller.CreateVuln)
		r.POST("/updateVuln", vuln_controller.UpdateVuln)
		r.POST("/deleteVuln", vuln_controller.DeleteVuln)
		r.GET("/getVuln", vuln_controller.GetVuln)

		//审核漏洞
		r.POST("/auditVuln", vuln_controller.AuditVuln)

	}
}
